lua/entities/gmod_wire_expression2/base/parser.lua

```diff
@@ -973,8 +973,9 @@ function Parser:Expr15()
 				else
 					self:Error("Operator (" .. v[1] .. ") must be followed by variable")
 				end
-			end ---@cast ident Token # Know it isn't nil from above check
-
+			else ---@cast ident Token # Know it isn't nil from above check
+				if v[1] == "$" then self.delta_vars[ident.value] = true end
```

Comment:
This behavior was moved from the parser to compiler. This shouldn't be fixing anything and if it is, whatever was depending on it should be moved to use the compiler's output

Reply:
Was partially wrong about this part, kept it in the parser only because we still don't have an analyzer pass, and compiler needs to know if a variable is ever used with $ even before the $ operator is used.. think I'll just stick with the parser for now until I get to said analyzer..